src/Opaleye/Internal/PrimQuery.hs

```diff
@@ -53,6 +53,7 @@ data PrimQuery' a = Unit
                               (Bindings HPQ.PrimExpr)
                               (PrimQuery' a)
                               (PrimQuery' a)
+                  | Antijoin  (PrimQuery' a) (PrimQuery' a)
```

Comment:
It might be possible, but antijoins don't create new bindings so I'm not sure they fit the shape of `Join` - but I'll give it a try tomorrow and see.